File: examples/src/example_readings_spring_boot_web_app/src/main/java/arena/examples/readings/springboot/CreateReadingRequest.java
package arena.examples.readings.springboot;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CreateReadingRequest(
    @JsonProperty("user_name") @NotBlank String userName,
    int value,
    String comment,
    @JsonProperty("device_id") @NotNull Long deviceId) {}

File: examples/src/example_readings_aspnet_web_app/Models/CreateReadingRequest.cs
namespace ArenaExamples.Readings.Aspnet.Models;

public class CreateReadingRequest
{
    public string UserName { get; set; } = default!;
    public int Value { get; set; }
    public string? Comment { get; set; }
    public int DeviceId { get; set; }
}
